document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutor.java

```diff
@@ -0,0 +1,99 @@
+package org.hypertrace.core.documentstore.mongo;
+
+import static java.util.function.Predicate.not;
+import static org.hypertrace.core.documentstore.mongo.MongoPaginationHelper.applyPagination;
+import static org.hypertrace.core.documentstore.mongo.MongoPaginationHelper.getLimitClause;
+import static org.hypertrace.core.documentstore.mongo.MongoPaginationHelper.getSkipClause;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoFilteringExpressionParser.getFilter;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoFilteringExpressionParser.getFilterClause;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoGroupingExpressionParser.getGroupClause;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoSelectingExpressionParser.getProjectClause;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoSelectingExpressionParser.getSelections;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoSortingExpressionParser.getOrders;
+import static org.hypertrace.core.documentstore.mongo.parser.MongoSortingExpressionParser.getSortClause;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCursor;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.conversions.Bson;
+import org.hypertrace.core.documentstore.query.PaginationDefinition;
+import org.hypertrace.core.documentstore.query.Query;
+
+@Slf4j
+public class MongoQueryExecutor {
+
+  public static Iterator<BasicDBObject> find(
+      final Query query, final com.mongodb.client.MongoCollection<BasicDBObject> collection) {
```

Comment:
Can we make these methods `non-static` and `collection will be its instance as mostly they will be used from the implementation of `Collection` interface?